python/arcticdb/adapters/lmdb_library_adapter.py

```diff
@@ -17,22 +20,93 @@
 from arcticdb.config import _DEFAULT_ENV
 from arcticdb.version_store._store import NativeVersionStore
 from arcticdb.adapters.arctic_library_adapter import ArcticLibraryAdapter, set_library_options
-from arcticdb_ext.storage import StorageOverride
+from arcticdb_ext.storage import StorageOverride, LmdbOverride
 from arcticdb.encoding_version import EncodingVersion
+from arcticdb.exceptions import ArcticNativeNotYetImplemented, LmdbOptionsError
 
 
 def _rmtree_errorhandler(func, path, exc_info):
     log.warn("Error removing LMDB tree at path=[{}]", path, exc_info=exc_info)
 
 
+@dataclass
+class ParsedQuery:
+    map_size: Optional[int] = None
+
+
+SUFFIX_TO_SCALE = {"KB": int(1e3), "MB": int(1e6), "GB": int(1e9), "TB": int(1e12)}
+
+
+MAP_SIZE_OPTION_ERROR_TEMPLATE = (
+    "Incorrect format for map_size option in LMDB connection string. Correct format is "
+    "a positive integer followed by one of (KB, MB, GB, TB), eg '200MB' or '10GB'. But option "
+    "map_size was [{}]."
+)
+
+
+def convert_size_str_to_bytes(size: str) -> int:
+    """
+    1MB -> int(1e6)
+    2GB -> int(2e9)
+    3TB -> int(3e12)
+    1mb -> raise
+    1Gb -> raise
+    """
+    suffix = size[-2:]
+    if suffix not in SUFFIX_TO_SCALE:
+        raise LmdbOptionsError(MAP_SIZE_OPTION_ERROR_TEMPLATE.format(size))
+    scale = SUFFIX_TO_SCALE[suffix]
+    mantissa_str = size[:-2]
+    try:
+        mantissa = int(mantissa_str)
+        if mantissa < 1:
+            raise ValueError()
+    except ValueError:
+        raise LmdbOptionsError(MAP_SIZE_OPTION_ERROR_TEMPLATE.format(size))
+    return mantissa * scale
+
+
+def parse_query(query: str) -> ParsedQuery:
+    if query and query.startswith("?"):
+        query = query.strip("?")
+    elif not query:
+        return ParsedQuery()
+
+    parsed_query = re.split("[;&]", query)
+    parsed_query = {t.split("=", 1)[0]: t.split("=", 1)[1] for t in parsed_query}
+
+    result = dict()
+    for key in parsed_query.keys():
+        field_dict = {field.name: field for field in fields(ParsedQuery)}
```

Comment:
its not a big deal but shouldn't this be outside of the for loop?
I don't think we need to recreate the field_dict on every iteration